sim/team-validator.ts

```diff
@@ -1767,6 +1775,13 @@ export class TeamValidator {
 		}
 		// Event-related ability restrictions only matter if we care about illegal abilities
 		const ruleTable = this.ruleTable;
+		if (ruleTable.has('obtainablemoves')) {
+			const tradebackEligible = ruleTable.has('allowtradeback') || (dex.gen === 2 && species.gen === 1);
```

Comment:
The other uses of `ruleTable.has('allowtradeback')` don't do this so I assume that there is a reason this is done differently here.

Reply:
My logic was that if tradebacks were allowed then this issue of not being able to use moves from a past gen wouldn't be a problem. Though I guess that's under the assumption that `allowtradeback` was also meant for you to be able to bring the mon back up to the current gen, which I'm not sure is the case.

Reply:
I'm in agreement here with urkerab that it should be removed; that's more of a policy thing with allowtradeback in custom rules for modern gens.